utils/mil_tools/mil_misc_tools/terminal_input.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/env python
+import sys
+import tty
+import termios
+
+__author__ = 'David Soto'
+
+# Adapted from http://stackoverflow.com/questions/510357/python-read-a-single-character-from-the-user/21659588#21659588
+def get_ch():
+    '''
+    Gets a single character from stdin. Doesn't echo to screen
+    Catches CTRL-C and CTRL-E and raises exceptions.
+    '''
+    fd = sys.stdin.fileno()
+    old_settings = termios.tcgetattr(fd)
+    try:
+        tty.setraw(fd)
+        ch = sys.stdin.read(1)
+    finally:
+        termios.tcsetattr(fd, termios.TCSADRAIN, old_settings)
```

Comment:
I'm really surprised that Python doesn't have a simpler way to do this, but your Stackoverflow source checks out.